bundles/org.openhab.binding.salus/src/main/java/org/openhab/binding/salus/internal/rest/GsonMapper.java

```diff
@@ -0,0 +1,346 @@
+        var displayName = findOrNull(map, "display_name");
+        var value = findObjectOrNull(map, "value");
+
+        // parse `properties`
+        var list = map.entrySet().stream().filter(entry -> entry.getKey() != null)
+                .filter(entry -> !"name".equals(entry.getKey())).filter(entry -> !"base_type".equals(entry.getKey()))
+                .filter(entry -> !"read_only".equals(entry.getKey()))
+                .filter(entry -> !"direction".equals(entry.getKey()))
+                .filter(entry -> !"data_updated_at".equals(entry.getKey()))
+                .filter(entry -> !"product_name".equals(entry.getKey()))
+                .filter(entry -> !"display_name".equals(entry.getKey()))
+                .filter(entry -> !"value".equals(entry.getKey()))
+                .map(entry -> new Pair<>(requireNonNull(entry.getKey()).toString(), (Object) entry.getValue()))
+                .toList();
+        // this weird thing need to be done,
+        // because `Collectors.toMap` does not support value=null
+        // and in our case, sometimes the values are null
+        SortedMap<@NotNull String, @Nullable Object> properties = new TreeMap<>();
+        for (var entry : list) {
+            properties.put(entry.key, entry.value);
+        }
+        properties = unmodifiableSortedMap(properties);
+
+        return Optional.of(buildDeviceProperty(name, baseType, value, readOnly, direction, dataUpdatedAt, productName,
+                displayName, properties));
+    }
+
+    private DeviceProperty<?> buildDeviceProperty(String name, @Nullable String baseType, @Nullable Object value,
+            @Nullable Boolean readOnly, @Nullable String direction, @Nullable String dataUpdatedAt,
+            @Nullable String productName, @Nullable String displayName,
+            SortedMap<String, @Nullable Object> properties) {
+        if ("boolean".equalsIgnoreCase(baseType)) {
+            Boolean bool;
+            if (value == null) {
+                bool = null;
+            } else if (value instanceof Boolean) {
+                bool = (Boolean) value;
+            } else if (value instanceof Number) {
+                bool = ((Number) value).longValue() != 0;
+            } else if (value instanceof String) {
+                bool = parseBoolean((String) value);
+            } else {
+                logger.warn("Cannot parse boolean from [{}]", value);
+                bool = null;
+            }
+            return new DeviceProperty.BooleanDeviceProperty(name, readOnly, direction, dataUpdatedAt, productName,
+                    displayName, bool, properties);
+        }
+        if ("integer".equalsIgnoreCase(baseType)) {
+            Long longValue;
+            if (value == null) {
+                longValue = null;
+            } else if (value instanceof Long) {
+                longValue = (Long) value;
+            } else if (value instanceof Number) {
+                longValue = ((Number) value).longValue();
```

Comment:
```suggestion
            } else if (value instanceof Number valueAsNumber) {
                longValue = valueAsNumber.longValue();
```

Can you also check the other occurences for java 17 pattern matching